Main.hs

```diff
@@ -337,22 +338,33 @@ processPragma qn = getUniqueCompilerPragma pragmaName qn >>= \case
     return $ ClassPragma NoCode
   Just (CompilerPragma _ s) | s == "class" ->
     return $ ClassPragma YesCode
-  _                                        -> return DefaultPragma
-
+  Just (CompilerPragma _ s) | "deriving" `isPrefixOf` s ->
+    -- parse a deriving clause for a datatype by tacking it onto a
+    -- dummy datatype and then only keeping the deriving part
+    case Hs.parseDecl ("data X = X " ++ s) of
```

Comment:
That's quite a hack, but it seems src-exts does not provide its internal Happy parsers readily, so I guess this is fine.

Reply:
Yes :) It seemed the easiest way to do it as `Language.Haskell.Exts.Parser` parser only seems to have functions/instances for bigger lumps of AST.